### PR TITLE
[SPARK-27472] add user guide for binary file data source

### PR DESCRIPTION
**Pull request by mengxr**
## What changes were proposed in this pull request?

Add user guide for binary file data source.

Screen Shot 2019-04-28 at 10 21 26 PM

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)
(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

Please review http://spark.apache.org/contributing.html before opening a pull request.

### CODE REVIEW

File: docs/sql-data-sources.md [hunk 1]
```diff
@@ -54,4 +54,5 @@ goes into specific options that are available for the built-in data sources.
   * [Compatibility with Databricks spark-avro](sql-data-sources-avro.html#compatibility-with-databricks-spark-avro)
   * [Supported types for Avro -> Spark SQL conversion](sql-data-sources-avro.html#supported-types-for-avro---spark-sql-conversion)
   * [Supported types for Spark SQL -> Avro conversion](sql-data-sources-avro.html#supported-types-for-spark-sql---avro-conversion)
+* [Whole Binary Files](sql-data-sources-binaryFile.html)
```
Comment (mengxr) [inline docs/sql-data-sources.md]: Use `Whole Binary Files` to indicate that this is loading each file as a whole record.

**Comment by mengxr** [comment]
cc: @WeichenXu123 @HyukjinKwon @cloud-fan

**Review by cloud-fan — APPROVED** [review]

File: docs/sql-data-sources-binaryFile.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+{% highlight java %}
+
+spark.read().format("binaryFile").option("pathGlobFilter", "*.png").load("/path/to/data");
+
+{% endhighlight %}
+</div>
+<div data-lang="python" markdown="1">
+{% highlight python %}
+
+spark.read.format("pathGlobFilter").option("pathGlobFilter", "*.png").load("/path/to/data")
```
Comment (viirya) [inline docs/sql-data-sources-binaryFile.md]: pathGlobFilter  -> binaryFile

File: docs/sql-data-sources-binaryFile.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+It supports the following read option:
+<table class="table">
+  <tr><th><b>Property Name</b></th><th><b>Default</b></th><th><b>Meaning</b></th></tr>
+  <tr>
+    <td><code>pathGlobFilter</code></td>
+    <td>none (accepts all)</td>
+    <td>
+    An optional glob pattern to only include files with paths matching the pattern.
+    The syntax follows <code>org.apache.hadoop.fs.GlobFilter</code>.
+    It does not change the behaivor of partition discovery.
```
Comment (viirya) [inline docs/sql-data-sources-binaryFile.md]: typo: behaivor -> behavior

File: docs/sql-data-sources-binaryFile.md (hunk 2 above)
Comment (mengxr) [reply docs/sql-data-sources-binaryFile.md]: Thanks for the catch! Updated.

**Review by viirya — APPROVED** [review]

**Comment by mengxr** [comment]
Merged into master.

**Comment by HyukjinKwon** [comment]
Thanks for doing me. LGTM too.

**Pull request closed without merging**